rtloader/common/builtins/_util.c

```diff
@@ -123,14 +126,21 @@ PyObject *subprocess_output(PyObject *self, PyObject *args)
 
     PyGILState_STATE gstate = PyGILState_Ensure();
 
+    static char *keywords[] = { "command", "raise_on_empty", "env", NULL };
     // `cmd_args` is mandatory and should be a list, `cmd_raise_on_empty` is an optional
     // boolean. The string after the ':' is used as the function name in error messages.
-    if (!PyArg_ParseTuple(args, "O|O:get_subprocess_output", &cmd_args, &cmd_raise_on_empty)) {
+#ifdef DATADOG_AGENT_THREE
+#    define ARGS_FORMAT "O|O$O:get_subprocess_output"
+#elif defined(DATADOG_AGENT_TWO)
+#    define ARGS_FORMAT "O|OO:get_subprocess_output"
+#endif
+    if (!PyArg_ParseTupleAndKeywords(args, kw, ARGS_FORMAT, keywords, &cmd_args, &cmd_raise_on_empty, &cmd_env)) {
+#undef ARGS_FORMAT
```

Comment:
Could we define subprocess_output args format only once instead of using `undef` each time ?

Reply:
What do you mean by “each time” ?
`#` directives are processed by the pre-processor at compile time so that what the compiler sees as third argument of `PyArg_ParseTupleAndKeywords` is a string literal.
I put the `#undef` because I dislike having symbols or macros having a wider scope than needed, but I can remove it if you find it disturbing.

Reply:
Sorry, my comment wasn't really clear indeed. I was indeed thinking of defining it once outside the function code so it can be reused elsewhere if needed (just the `$O`/`O` part). And just concatenate it here.

WDYT ?

Reply:
Thanks for the clarification !
I just amended this PR and here is the change: https://github.com/DataDog/datadog-agent/pull/5987/commits/c2170474f80a57a6e3f68cc8e74a0976c396d60b

I think there are pros and cons for both ways.
TBH, I personally find weird to have a macro for the keyword-only arguments separator (`$`) and not for the optional arguments separator (`|`).
And I would find rather unreadable to replace all [the standard short format strings](https://docs.python.org/3/c-api/arg.html) by custom verbosely named macros.
I understand that this is made necessary by the fact that some formats (like `$`) are not supported by all python versions.
On the other hand, the change https://github.com/DataDog/datadog-agent/pull/5987/commits/c2170474f80a57a6e3f68cc8e74a0976c396d60b might increase the risk that, when we’ll drop the support of the deprecated Python 2, we’ll leave the macro, that will always be defined to `$`, but that will be useless at that point.

It’s hard to tell what will come first: a new use of the keyword-only separator (`$`) or the drop of Python 2.

Anyhow, the macro that conditionally defines `$` (https://github.com/DataDog/datadog-agent/pull/5987/commits/c2170474f80a57a6e3f68cc8e74a0976c396d60b) is now part of this PR.

Let me know if that’s looks good to you.